platforms/hyperledger-fabric/configuration/roles/create/ca-tools/orderer/tasks/main.yaml

```diff
@@ -115,25 +115,81 @@
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: network.env.proxy != 'none'  
+
+############################################################################################
+# This task copies the msp tlscacerts from vault
+- name: Copy the msp tlscacerts from vault
+  shell: |
+    vault kv get -field=tlscacerts {{ vault.secret_path | default('secretsv2') }}/crypto/ordererOrganizations/{{ component_name }}/users/admin/msp > ca-{{ component_name }}-{{ item.external_url_suffix }}-8443.pem
+    mkdir -p ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/tlscacerts/
+    mv ca-{{ component_name }}-{{ item.external_url_suffix }}-8443.pem ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/tlscacerts/
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: network.env.proxy != 'none'
+
+############################################################################################
+# This task copies the msp cacerts from vault proxy is none
+- name: Copy the msp cacerts from vault
+  shell: |
+    vault kv get -field=cacerts {{ vault.secret_path | default('secretsv2') }}/crypto/ordererOrganizations/{{ component_name }}/users/admin/msp > ca-{{ component_name }}-{{ item.services.ca.grpc.port }}.pem
+    mkdir -p ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/cacerts/
+    mv ca-{{ component_name }}-{{ item.services.ca.grpc.port }}.pem ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/cacerts/
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: network.env.proxy == 'none'  
+
+############################################################################################
+# This task copies the msp tlscacerts from vault when proxy is none
+- name: Copy the msp tlscacerts from vault
+  shell: |
+    vault kv get -field=tlscacerts {{ vault.secret_path | default('secretsv2') }}/crypto/ordererOrganizations/{{ component_name }}/users/admin/msp > ca-{{ component_name }}-{{ item.services.ca.grpc.port }}.pem
+    mkdir -p ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/tlscacerts/
+    mv ca-{{ component_name }}-{{ item.services.ca.grpc.port }}.pem ./build/crypto-config/ordererOrganizations/{{ component_name }}/msp/tlscacerts/
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: network.env.proxy == 'none'     
 
 ############################################################################################
-# This task copies the tls server.crt file from the respective CA Tools CLI 
+# This task copies the tls server.crt from vault
 # to the build directory
-- name: Copy the tls server.crt file from the ca tools
+- name: Copy the tls server.crt from vault
```

Comment:
Please rename the task and the above comment to `Fetch the tls server.crt from vault`